src/pools/D3MAaveDaiPool.sol

```diff
@@ -116,28 +118,30 @@ contract D3MAaveDaiPool is D3MPoolBase {
     }
 
     // --- Collect any rewards ---
-    function collect(address[] memory assets, uint256 amount) external auth returns (uint256 amt) {
+    function collect(address[] memory assets, uint256 amount) external returns (uint256 amt) {
         require(king != address(0), "D3MAaveDaiPool/king-not-set");
 
         amt = rewardsClaimer.claimRewards(assets, amount, king);
         emit Collect(king, assets, amt);
     }
 
     function transferShares(address dst, uint256 amt) external override returns (bool) {
-        return ShareTokenLike(share).transfer(dst, amt);
+        return ShareTokenLike(adai).transfer(dst, amt);
     }
 
+    function accrueIfNeeded() external override {}
+
     // --- Balance in standard ERC-20 denominations
     function assetBalance() external view override returns (uint256) {
-        return ShareTokenLike(share).balanceOf(address(this));
+        return ShareTokenLike(adai).balanceOf(address(this));
     }
 
     function shareBalance() external view override returns (uint256) {
-        return ShareTokenLike(share).balanceOf(address(this));
+        return ShareTokenLike(adai).balanceOf(address(this));
```

Comment:
just flagging that this required updating `Hub.quit` to use `pool.assetBalance` instead of `shareBalance` which I think fixed a mix up I had previously, just wanted to make sure it made sense to you especially with the comment below about shareTransfer